src/utils/viewer/viewer.cpp

```diff
@@ -118,80 +155,73 @@ class Viewer : public Mn::Platform::Application {
-  // assumes that it "owns" things owned by the scene manager
-  esp::scene::SceneManager sceneManager_;
+  // The simulator object backend for this viewer instance
+  std::unique_ptr<esp::sim::Simulator> simulator_;
 
-  std::shared_ptr<esp::physics::PhysicsManager> physicsManager_;
+  // The managers belonging to the simulator
+  std::shared_ptr<esp::assets::managers::ObjectAttributesManager>
+      objectAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::AssetAttributesManager>
+      assetAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::StageAttributesManager>
+      stageAttrManager_ = nullptr;
+  std::shared_ptr<esp::assets::managers::PhysicsAttributesManager>
+      physAttrManager_ = nullptr;
 
   bool debugBullet_ = false;
 
-  std::vector<int> sceneID_;
+  esp::scene::SceneNode* rootNode_ = nullptr;
   esp::scene::SceneNode* agentBodyNode_ = nullptr;
-  esp::scene::SceneNode* rgbSensorNode_ = nullptr;
-
-  std::string stageFileName;
-  esp::scene::SceneGraph* sceneGraph_;
-  esp::scene::SceneNode* rootNode_;
-
-  int navMeshVisPrimID_ = esp::ID_UNDEFINED;
-  esp::scene::SceneNode* navMeshVisNode_ = nullptr;
 
+  // Scene or stage file to load
+  std::string sceneFileName;
   esp::gfx::RenderCamera* renderCamera_ = nullptr;
-  esp::nav::PathFinder::ptr pathfinder_;
-  esp::scene::ObjectControls controls_;
-  std::vector<int> objectIDs_;
-
   bool drawObjectBBs = false;
 
   Mn::Timeline timeline_;
 
   Mn::ImGuiIntegration::Context imgui_{Mn::NoCreate};
   bool showFPS_ = true;
-  bool frustumCullingEnabled_ = true;
 
   // NOTE: Mouse + shift is to select object on the screen!!
   void createPickedObjectVisualizer(unsigned int objectId);
   std::unique_ptr<ObjectPickingHelper> objectPickingHelper_;
 };
 
 Viewer::Viewer(const Arguments& arguments)
-    : Mn::Platform::Application{arguments,
-                                Configuration{}
-                                    .setTitle("Viewer")
-                                    .setWindowFlags(
-                                        Configuration::WindowFlag::Resizable),
-                                GLConfiguration{}
-                                    .setColorBufferSize(
-                                        Mn::Vector4i(8, 8, 8, 8))
-                                    .setSampleCount(4)},
-      pathfinder_(esp::nav::PathFinder::create()),
-      controls_() {
+    : Mn::Platform::Application{
+          arguments,
+          Configuration{}.setTitle("Viewer").setWindowFlags(
+              Configuration::WindowFlag::Resizable),
+          GLConfiguration{}
+              .setColorBufferSize(Mn::Vector4i(8, 8, 8, 8))
+              .setSampleCount(4)} {
   Cr::Utility::Arguments args;
 #ifdef CORRADE_TARGET_EMSCRIPTEN
   args.addNamedArgument("scene")
 #else
   args.addArgument("scene")
 #endif
-      .setHelp("scene", "scene file to load")
+      .setHelp("scene", "scene/stage file to load")
```

Comment:
Nit: maybe it is better to display `scene (stage)`

Reply:
From an API perspective we treat these the same way for a Simulator. 
Once dev is complete you'll be able to provide a stage asset, stage config, or scene config to initialize the Simulator. If you only provide the stage (current implementation), then we'll interpret that as a scene which is empty except for the stage you provide.

Reply:
maybe `.setHelp("scene", "scene or stage file to load.")` ?

Reply:
> From an API perspective we treat these the same way for a Simulator.
Exactly. I agree.

> .setHelp("scene", "scene or stage file to load.")
I see. They are actually different. When I say `scene (stage)`. I thought they were the same.
Then you do not have to change. 
The question is do we need to add comments somewhere so the user knows the difference in between?

Reply:
This should probably be documented across the stack when the scene import feature arrives. We're planning a docs page to cover this. May also want to put up a terminology docs page to support naming conventions.